react/next/utils/next.ts

```diff
@@ -0,0 +1,70 @@
+import { SliderState, SliderProps } from '../typings'
+
+interface NextSlides {
+  nextSlides: number | undefined
+  nextPosition: number | undefined
+}
+
+/**
+ * Populate the next slides
+ * TODO: Implement the 'fake' translate when in infinite mode when right end is reached
+ * @param state : slider current state
+ * @param props : slider props
+ * @param slidesHavePassed : slides that passed
+ */
+const populateNextSlides = (
```

Comment:
Couldn't you extract all math logic here? IMO it's spread all over the PR, and it's hard to understand. If, just like Thanos, it's inevitable, hide its implementation details in one place and just call it